src/controller/tests/data_model/TestRead.cpp

```diff
@@ -1769,10 +1769,31 @@ void TestReadInteraction::TestReadHandler_MultipleSubscriptions(nlTestSuite * ap
     //
     for (size_t i = 0; i < (app::InteractionModelEngine::kReadHandlerPoolSize + 1); i++)
     {
-        NL_TEST_ASSERT(apSuite,
-                       Controller::SubscribeAttribute<Clusters::UnitTesting::Attributes::ListStructOctetString::TypeInfo>(
-                           &ctx.GetExchangeManager(), sessionHandle, kTestEndpointId, onSuccessCb, onFailureCb, 0, 20,
-                           onSubscriptionEstablishedCb, nullptr, false, true) == CHIP_NO_ERROR);
+        Controller::detail::ReportAttributeParams<Clusters::UnitTesting::Attributes::ListStructOctetString::TypeInfo::DecodableType>
+            params(sessionHandle);
+
+        params.mOnReportCb                  = onSuccessCb;
+        params.mOnErrorCb                   = onFailureCb;
+        params.mOnSubscriptionEstablishedCb = onSubscriptionEstablishedCb;
+        params.mOnResubscriptionAttemptCb   = nullptr;
+        params.mOnDoneCb                    = nullptr;
+        params.mMinIntervalFloorSeconds     = 0;
+        params.mMaxIntervalCeilingSeconds   = 20;
+        params.mReportType                  = app::ReadClient::InteractionType::Subscribe;
+        params.mKeepSubscriptions           = true;
+        params.mIsFabricFiltered            = false;
+
+        // Increase timeout, so that slow versions (specifically gcc_debug) does not time out.
+        params.mTimeout = chip::System::Clock::Seconds16(15);
+
+        CHIP_ERROR err =
+            Controller::detail::ReportAttribute<Clusters::UnitTesting::Attributes::ListStructOctetString::TypeInfo::DecodableType>(
```

Comment:
You should not be using "detail" APIs here.  There are public APIs for all this; please use them.